LandSandBoat/server
# [LUA] Create job_utils/thief, migrate JA functionality, and add Larceny

**_I affirm:_**

## What does this pull request do?



- Create job_utils/thief.lua and populates necessary ability check and ability use functions.
- Moves functionality from all Thief globals/abilities files to globals/job_utils/thief.lua.
- Sets up Larceny to steal status effects, prioritizing SP Abilities used by enemies.
- Adjusts the Larceny ability to use the correct actions and messages when successful or failing to steal a buff.

## References

https://www.bg-wiki.com/ffxi/Larceny

## Steps to test these changes



1. Upload changes to your server.
2. Load the server.
3. Use the !changejob THF 99 command.
4. Use all JA's to confirm they do not throw errors.
5. Use Larceny against a mob that uses SP Abilities and has other buffs to test stealing priority.
